providers/core/core.xhr.js

```diff
@@ -0,0 +1,164 @@
+  }
+  return PromiseCompat.resolve();
+};
+
+XhrProvider.prototype.abort = function() {
+  "use strict";
+  this._xhr.abort();
+  return PromiseCompat.resolve();
+};
+
+XhrProvider.prototype.getResponseHeader = function(header) {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.getResponseHeader(header));
+};
+
+XhrProvider.prototype.getAllResponseHeaders = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.getAllResponseHeaders());
+};
+
+XhrProvider.prototype.setRequestHeader = function(header, value) {
+  "use strict";
+  this._xhr.setRequestHeader(header, value);
+  return PromiseCompat.resolve();
+};
+
+XhrProvider.prototype.overrideMimeType = function(mime) {
+  "use strict";
+  this._xhr.overrideMimeType(mime);
+  return PromiseCompat.resolve();
+};
+
+XhrProvider.prototype.getReadyState = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.readyState);
+};
+
+XhrProvider.prototype.getResponse = function() {
+  "use strict";
+  if (this._xhr.response === null) {
+    return PromiseCompat.resolve(null);
+  } else if (this._xhr.responseType === "text" || this._xhr.responseType === "") {
+    return PromiseCompat.resolve({ string: this._xhr.response });
+  } else if (this._xhr.responseType === "arraybuffer") {
+    return PromiseCompat.resolve({ buffer: this._xhr.response });
+  } else if (this._xhr.responseType === "json") {
+    return PromiseCompat.resolve({ object: this._xhr.response });
+  }
+
+  return PromiseCompat.reject("core.xhr cannot determine type of response");
+};
+
+XhrProvider.prototype.getResponseText = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.responseText);
+};
+
+XhrProvider.prototype.getResponseURL = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.responseURL);
+};
+
+XhrProvider.prototype.getResponseType = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.responseType);
+};
+
+XhrProvider.prototype.setResponseType = function(type) {
+  "use strict";
+  this._xhr.responseType = type;
+  return PromiseCompat.resolve();
+};
+
+XhrProvider.prototype.getStatus = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.status);
+};
+
+XhrProvider.prototype.getStatusText = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.statusText);
+};
+
+XhrProvider.prototype.getTimeout = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.timeout);
+};
+
+XhrProvider.prototype.setTimeout = function(timeout) {
+  "use strict";
+  this._xhr.timeout = timeout;
+  return PromiseCompat.resolve();
+};
+
+XhrProvider.prototype.getWithCredentials = function() {
+  "use strict";
+  return PromiseCompat.resolve(this._xhr.withCredentials);
+};
+
+XhrProvider.prototype.setWithCredentials = function(wc) {
+  "use strict";
+  this._xhr.withCredentials = wc;
+  return PromiseCompat.resolve();
+};
+
+
+XhrProvider.prototype._setupListeners = function() {
+  "use strict";
```

Comment:
you should add an 'onClose' listener to remove / abort the underlying _xhr when closed, so that instances get fully freed when closed by the consumer.

Example:
https://github.com/freedomjs/freedom/blob/master/providers/core/core.view.js#L22